### PR TITLE
More explanatory text.. Fixes #587. Fixes #591

### LINKED ISSUE #591: Can we clarify the Misconfiguration section?
I am really struggling with this section and before I try again to make a proposal (I failed 3 times on my side) I will ask questions here and then see if and how I can make a proposal. Again consider WHO is the target audience of this section, not TLS developers or people who have been 30 years in it. Operational people will be either completely lost OR worse they will believe they understood which won't be the case. 

I understand too we can't expand too much the text here but now it is reaaally too compact to read it from non-expert side. 

So to guide the help I need, here are a few questions.

In the clause 

```
"The retry mechanism repairs inconsistencies, provided the server is
authoritative for the public name. If server and advertised keys mismatch, the
server will reject ECH and respond with "retry_configs". If the server does
not understand
the "encrypted_client_hello" extension at all, it will ignore it as required by
{{Section 4.1.2 of RFC8446}}. Provided the server can present a certificate
valid for the public name, the client can safely retry with updated settings,
as described in {{rejected-ech}}."
```
Can somebody provide me with a precise definition of the term `the server is authoritative for the public name`

Can somebody clarify me `If server and advertised keys mismatch` and give me some example. Good example of where you have to build in your brain, which server, which keys, why can they mismatch, etc. remember not everybody is a TLS developer or 30 years in it. Especially for WHO is the audience of this section.

Can somebody clarify me `can present a certificate valid for the public name`. 

Finally can someone give an example after 
```
Unless ECH is disabled as a result of successfully establishing a connection to
the public name 

```

### PR DESCRIPTION
**Pull request by ekr**
Rewrite the overview section to be a bit clearer and clarified the misconfiguration pieces.

* Added more narrative text around the topologies.
* Moved much of misconfiguration piece up to overview so people have some signposting.
* Expanded the misconfiguration pieces a bit.

### CODE REVIEW

**Comment by ekr** [comment]
@dennisjackson @martinthomson

File: draft-ietf-tls-esni.md [hunk 1]
```diff
@@ -1232,28 +1278,35 @@ guarantee. Thus this protocol was designed to be robust in case of
...
+allow a network attacker to disclose the contents of this ClientHello,
+including the SNI. This restriction may mean that certain kinds of
+network misbehavior or misconfiguration (see {{middleboxes}} may make
+it impossible to complete an ECH connection where an ordinary TLS 1.3
+connection would work.
+
+Clients MAY attempt to use another server or ECHConfig value from the
 DNS results, if one is provided.
 
+
```
Comment (wbl) [inline draft-ietf-tls-esni.md:1309]: Looks spurious

File: draft-ietf-tls-esni.md [hunk 2]
```diff
@@ -249,6 +249,52 @@ anonymity set are indistinguishable from one another. Moreover, it should
...
+may occur, for instance, from DNS misconfiguration, caching issues, or an
+incomplete rollout in a multi-server deployment. This may also occur if a server
+loses its ECH keys, or if a deployment of ECH must be rolled back on the server.
+Whatever the cause, there are two main consequences:
+
+* The server has ECH configured but the client has an ECH configuration with a
+  key not known to the server.
+
+* The server has ECH configured but the client has a ECH configuration
+  and so is attempting ECH.
```
Comment (sftcd) [inline draft-ietf-tls-esni.md:265]: Missing a negative in 2nd bullet above?

**Comment by sftcd** [comment]
other than the above, this change seems fine

File: draft-ietf-tls-esni.md [hunk 3]
```diff
@@ -249,6 +249,52 @@ anonymity set are indistinguishable from one another. Moreover, it should
 achieve this goal without affecting any existing security properties of TLS 1.3.
 See {{goals}} for more details about the ECH security and privacy goals.
 
+
+### Misconfiguration and Deployment Concerns {#misconfiguration}
+
+It is possible for ECH advertisements and servers to become inconsistent. This
+may occur, for instance, from DNS misconfiguration, caching issues, or an
+incomplete rollout in a multi-server deployment. This may also occur if a server
+loses its ECH keys, or if a deployment of ECH must be rolled back on the server.
```
Comment (martinthomson) [inline draft-ietf-tls-esni.md:258]:
```suggestion
loses its ECH keys, or if a deployment of ECH is rolled back on the server.
```
or "needs to be", but "is" works.

File: draft-ietf-tls-esni.md (hunk 2 above)
Comment (martinthomson) [inline draft-ietf-tls-esni.md:265]:
```suggestion
* The server does not have ECH configured but the client has a ECH configuration
  and so is attempting ECH.
```

File: draft-ietf-tls-esni.md [hunk 4]
```diff
@@ -1232,28 +1278,35 @@ guarantee. Thus this protocol was designed to be robust in case of
...
+### Misconfiguration and Deployment Concerns {#misconfiguration-deployment}
 
-It is possible for ECH advertisements and servers to become inconsistent. This
-may occur, for instance, from DNS misconfiguration, caching issues, or an
-incomplete rollout in a multi-server deployment. This may also occur if a server
-loses its ECH keys, or if a deployment of ECH must be rolled back on the server.
+The retry mechanism described in {{rejected-ech}} allows the server to
+recover from situations where it is not able to decrypt the ECH field,
+for instance if the client has a stale ECHConfig or the server has lost
+the keys. The security of this mechanism is rooted in the server having
```
Comment (martinthomson) [inline draft-ietf-tls-esni.md:1286]:
```suggestion
the keys. The security of this mechanism depends on the server having
```

File: draft-ietf-tls-esni.md [hunk 5]
```diff
@@ -1232,28 +1278,35 @@ guarantee. Thus this protocol was designed to be robust in case of
...
-Unless ECH is disabled as a result of successfully establishing a connection to
-the public name, the client MUST NOT fall back to using unencrypted
-ClientHellos, as this allows a network attacker to disclose the contents of this
-ClientHello, including the SNI. It MAY attempt to use another server from the
+- If there is a transport-level failure
+- If TLS handshake does not complete successfully
+- If TLS handshake completes using the ClientHelloOuter but with a certificate
+  which is not valid for the public name.
+
+Falling back to unencrypted ClientHellos in any of these cases would
```
Comment (martinthomson) [inline draft-ietf-tls-esni.md:1299]:
```suggestion
Falling back to an unencrypted ClientHello in any of these cases would
```

**Review by martinthomson — APPROVED** [review]
Any point saying that:

> This makes availability of a certificate for the public name - and its associated private key - a critical component for server deployment of ECH.  This certificate is used both to recover from transient misconfiguration problems and to disable ECH in the event that a server cannot continue to support the capability.

File: draft-ietf-tls-esni.md [hunk 6]
```diff
@@ -249,6 +249,52 @@ anonymity set are indistinguishable from one another. Moreover, it should
...
+It is possible for ECH advertisements and servers to become inconsistent. This
+may occur, for instance, from DNS misconfiguration, caching issues, or an
+incomplete rollout in a multi-server deployment. This may also occur if a server
+loses its ECH keys, or if a deployment of ECH must be rolled back on the server.
+Whatever the cause, there are two main consequences:
+
+* The server has ECH configured but the client has an ECH configuration with a
+  key not known to the server.
+
+* The server has ECH configured but the client has a ECH configuration
```
Comment (ekr) [inline draft-ietf-tls-esni.md:264]:
```suggestion
* The server does not have ECH configured but the client has a  ECH configuration
```

File: draft-ietf-tls-esni.md (hunk 1 above)
Comment (ekr) [inline draft-ietf-tls-esni.md:1309]:
```suggestion
```

**Pull request merged**